### ISSUE TITLE
Setting particular hero spells to quickcast causes all that hero's default spell hotkeys to be lost next time Dota2 is opened

### ISSUE THREAD

**Issue by ennever** [jeffhill/Dota2#273]
### Description

Here are the steps:
- Enable Advanced Quickcast/Autocast Hotkeys
- In-game for a hero with the default hotkeys, set one of your hero's spells to Quickcast, leaving the rest on regular cast and their default hotkeys (this can be done in demo mode)
- The game will play as normal with the default hotkeys still viable
- Close Dota2
- When Dota2 is reopened, the hero's unchanged spells all have no hotkey and you will manually need to change them back

In the match ID I listed it was my second Viper game after changing his ultimate to Quickcast on R, and you can see me open the settings and fix it by manually setting QWE hotkeys around 30 seconds before the first creepwave spawn. 

### Example Match ID (and possibly Timestamp)

6596144579

**Comment by anonymo0se** [comment]
https://www.reddit.com/r/DotA2/comments/u7y39w/hotkeys_get_unbound_every_time_i_start_dota/

I found a solution to this here.

**Comment by ennever** [comment]
Yeah, it's a workaround (setting the default spells explicitly to "cast" when you change one to quickcast), but it is super confusing still and not at all obvious. The first time it happened I was so confused because I couldn't even use my hotkeys to level the spells anymore level 1, and that got me killed at the bounty rune fight.

**Comment by nluna9** [comment]
This!!!!!!!! Is so frustrating!

**Comment by Venryx** [comment]
Yeah I hit this same issue years ago.

The TL;DR for the workaround: If you ever are changing an ability's hotkey setting, **you must re-set all the other ability hotkeys at the same time**, because only the set of hotkeys "modified during that change-set" end up existing in your cloud storage after DOTA is restarted.
